### PR TITLE
fix!: examine response content-type if no contentType is set

### LINKED ISSUE #226: `Content-Type` response header is ignored
Related to https://github.com/googleapis/google-api-nodejs-client/issues/1777.  

We accept a `responseType` as part of the request config that dictates what processing we do with the response body before returning it in `res.data`.  By default, we just assume `json` if not told otherwise.  The problem pops up when the content type is defined in a response header, and we just ignore it.  When an `application/gzip` content type comes back in a header, I'd argue we shouldn't try `JSON.parse`-ing that by default :)

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/gaxios/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #226 🦕

### CODE REVIEW

File: src/gaxios.ts [hunk 1]
```diff
@@ -271,7 +271,7 @@ export class Gaxios {
     }
 
     opts.validateStatus = opts.validateStatus || this.validateStatus;
-    opts.responseType = opts.responseType || 'json';
+    opts.responseType = opts.responseType || 'unknown';
```
Comment (ddelgrosso1) [inline src/gaxios.ts:274]: I want to get some opinions on whether or not we would consider this a breaking change. Previously we defaulted for to JSON as the `contentType` if the caller did not specify one. This is sort of a faulty assumption and we probably should have been looking at the response headers. I changed this to default to unknown but I fear that consumers may have been relying on this old JSON behavior.

**Comment by ddelgrosso1** [comment]
Compodoc > 1.1.19 has a dependency on marked 5.0.2 which requires node 18 minimum. Pinning compodoc in this PR.

File: src/gaxios.ts (hunk 1 above)
Comment (danielbankhead) [reply src/gaxios.ts:274]: I think this would technically be breaking, although customers would be relying on a bug.

Comment (sofisl) [reply src/gaxios.ts:274]: If we do this, we should also update the README: https://github.com/googleapis/gaxios#request-options

**Review by sofisl — APPROVED** [review]

**Pull request merged**